test/modules/finance.spec.ts

```diff
@@ -313,17 +324,103 @@ describe('finance', () => {
           expect(bitcoinAddress).toBeTypeOf('string');
-          expect(bitcoinAddress).toSatisfy(isValidBtcAddress);
-        });
+          expect(bitcoinAddress).toSatisfy(isBtcAddress);
+        });
+        it.each`
+          type         | expectedPrefix
+          ${'legacy'}  | ${'1'}
+          ${'segwit'}  | ${'3'}
+          ${'bech32'}  | ${'bc1'}
+          ${'taproot'} | ${'bc1p'}
+        `(
+          'should handle the type = $type argument',
+          ({
+            type,
+            expectedPrefix,
+          }: {
+            type: BitcoinAddressType;
+            expectedPrefix: string;
+          }) => {
+            const bitcoinAddress = faker.finance.bitcoinAddress({
+              type,
+            });
+
+            expect(bitcoinAddress).toBeTruthy();
+            expect(bitcoinAddress).toBeTypeOf('string');
+            expect(bitcoinAddress).toSatisfy(isBtcAddress);
+            expect(bitcoinAddress.startsWith(expectedPrefix)).toBe(true);
+          }
+        );
+
+        it.each`
+          network      | expectedPrefixes
+          ${'mainnet'} | ${['1', '3', 'bc1', 'bc1p']}
+          ${'testnet'} | ${['2', 'm', 'tb1', 'tb1p']}
+        `(
+          'should handle the network = $network argument',
+          ({
+            network,
+            expectedPrefixes,
+          }: {
+            network: BitcoinNetwork;
+            expectedPrefixes: string[];
+          }) => {
+            const bitcoinAddress = faker.finance.bitcoinAddress({
+              network,
+            });
+
+            expect(bitcoinAddress).toBeTruthy();
+            expect(bitcoinAddress).toBeTypeOf('string');
+            expect(bitcoinAddress).toSatisfy(isBtcAddress);
+            expect(
+              expectedPrefixes.some((p: string) => bitcoinAddress.startsWith(p))
+            ).toBe(true);
+          }
+        );
+
+        it.each`
+          type         | network      | expectedPrefix
+          ${'legacy'}  | ${'mainnet'} | ${'1'}
+          ${'legacy'}  | ${'testnet'} | ${'m'}
+          ${'segwit'}  | ${'mainnet'} | ${'3'}
+          ${'segwit'}  | ${'testnet'} | ${'2'}
+          ${'bech32'}  | ${'mainnet'} | ${'bc1'}
+          ${'bech32'}  | ${'testnet'} | ${'tb1'}
+          ${'taproot'} | ${'mainnet'} | ${'bc1p'}
+          ${'taproot'} | ${'testnet'} | ${'tb1p'}
+        `(
+          'should handle the type = $type and network = $network arguments',
+          ({
+            type,
+            network,
+            expectedPrefix,
+          }: {
+            type: BitcoinAddressType;
+            network: BitcoinNetwork;
+            expectedPrefix: string;
+          }) => {
+            const bitcoinAddress = faker.finance.bitcoinAddress({
+              type,
+              network,
+            });
+
+            expect(bitcoinAddress).toBeTruthy();
+            expect(bitcoinAddress).toBeTypeOf('string');
+            expect(bitcoinAddress.startsWith(expectedPrefix)).toBe(true);
```

Comment:
This might produce better error messages, if we move the check into the satisfy check:

```suggestion
            expect(bitcoinAddress).toSatisfy((v) -> v.startsWith(expectedPrefix));
```

But that is optional IMO.